crt0/src/crt0_BASIC.s

```diff
@@ -0,0 +1,131 @@
+;   MA 02110-1301, USA.
+;
+;  As a special exception, if you link this library with other files,
+;  some of which are compiled with SDCC, to produce an executable,
+;  this library does not by itself cause the resulting executable to
+;  be covered by the GNU General Public License. This exception does
+;  not however invalidate any other reasons why the executable file
+;   might be covered by the GNU General Public License.
+;--------------------------------------------------------------------------
+
+; Memory layout:
+;  0x8000: RAM at addresses lower than the load-address:
+;    -> Use 0x8000--0x95ff, 5.5 KiB, as work-ram
+;    -> Use 0x9600--0x9700, 257 bytes, as the interrupt vector table.
+;  0x9800: Destination address where BASIC loads our code to.
+;    -> Continues to 0xc7ff for BASIC IIIa (12 KiB)
+;    -> Continues to 0xffff for BASIC IIIb (26 KiB)
+;  0x9898: The interrupt handler, pointed to by the interrupt vector table.
+;
+; Note: The NMI (reset button) is not available to the user application.
+
+  .module crt0
+  .globl  _main
+
+  .area _HEADER (ABS)
+  ;; Destination address used by the BASIC `LOAD` command
+  .org  0x9800
+init:
+  di                ; disable interrupt
+  im 2              ; interrupt mode 2 (this will not change)
+
+  ld sp,#0x95f0     ; set stack pointer at end of our 5.5k block of RAM
+  xor a             ; clear RAM (to value 0x00)
+  ld hl,#0x8000     ;   by setting value 0
+  ld (hl),a         ;   to $c000 and
+  ld de,#0x8001     ;   copying (LDIR) it to next byte
+  ld bc,#0x1600-17  ;   for 5.5 KB minus 17 bytes
+  ldir              ;   do that
+
+  ;; ensure this runs fine on SC-3000 too
+  ld a,#0x92
+  out (0xDF),a      ; Config PPI (no effect on SG-1000)
+  ld a,#7
+  out (0xDE),a      ; Select ROW 7 (row 7 of PPI is joypad = default - no effect on SG-1000)
+
+  ;; Set up for interrupt-mode 2
+  ;; Register I controls the vector MSB. The vector LSB is whatever happens to be on the bus.
+  ;; So, a 257-entry table with a single repeated byte (0x98) will give us a reliable 0x9898 call.
+  ld a,#0x96        ; Vector table at 0x9600, between work-ram and user code.
+  ld i,a
+
+  ld a,#0x98
+  ld hl,#0x9600
+  ld (hl),a         ; Initialize first byte of vector table at 0x9600 to value 0x98
+  ld de,#0x9601     ; Prepare copy to the next byte
+  ld bc,#0x0100     ; Copy for 256 bytes, giving 257 bytes total.
+  ldir
+
+  ;; Initialise global variables
+  call    gsinit
+  call    _SG_init
+  ei                ; re-enable interrupts before going to main()
+  call  _main
+  jp  _exit
+
+  .org    0x9898    ; handle IRQ
+  jp _SG_isr
+
+  ;; Ordering of segments for the linker.
+  .area _HOME
+  .area _CODE
+  .area _INITIALIZER
+  .area   _GSINIT
+  .area   _GSFINAL
+
+  .area _DATA
+  .area _INITIALIZED
+  .area _BSEG
+  .area   _BSS
+  .area   _HEAP
+
+  .area   _CODE
+__clock::
+  ld  a,#2
+  rst     0x08
+  ret
+
+_exit::
+  ;; Exit - special code to the emulator
+  ld a,#0
+  rst 0x08
```

Comment:
I suspect this part contains leftovers from crt0_sg.s - which were already leftover there...

Reply:
Can you try if it works by removing this part above and and moving the 
```
1$:
  halt
  jr 1$
```
part right below the `call _main` (at row 85) where the `jp _exit` (row 86) currently sits?